fix(core): run afterRender callbacks outside of the Angular zone

devknoll opened the pull request:
afterRender should run outside of the Angular zone so that it does not trigger further CD cycles

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
Callbacks are executed in the Angular zone, causing patched APIs like `setTimeout` to trigger subsequent change detection cycles, and further `afterRender` callbacks.

Issue Number: #51343


## What is the new behavior?
Callbacks are always executed outside of the Angular zone, preventing accidental infinite render loops.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


`afterRender` is in developer preview, and so this is not considered a breaking change.


## Other information

Closes #51343

devknoll commented:
cc @kara

alxhub commented:
@devknoll can you change the commit to `fix`, not `feat`? Our merge tooling will reject merging a `feat` commit to the `16.2.x` branch.

devknoll committed:
fix(core): run afterRender callbacks outside of the Angular zone

afterRender should run outside of the Angular zone so that it does not trigger further CD cycles

jessicajaniuk commented:
@devknoll I see this has `action: presubmit`, but this is showing as passing. Is another presubmit needed? Or a TGP?

kara commented:
@jessicajaniuk oh, my mistake. I added the label bc I didn't see it already ran. Will remove.

jessicajaniuk commented:
@devknoll I'm going to update this to `target: minor`, but we'll need a patch PR to get it into `16.2.x`. Please create that at your leisure.

jessicajaniuk commented:
This PR was merged into the repository by commit 3a19d6b7437e1812ae70b3784fd6a8a185b330b1.
